src/ploigos_step_runner/step_implementers/automated_governance/rekor.py

```diff
@@ -0,0 +1,277 @@
+------------------|-----------|----------------|-----------
+`rekor-server`     | True |         | Version to use when building the container image
+
+Result Artifacts
+----------------
+Results artifacts output by this step.
+
+Result Artifact Key | Description
+--------------------------|------------
+`container-image-version` | Container version to tag built image with
+`image-tar-file`          | Path to the built container image as a tar file
+`image-tar-hash`          | Path to the built container image as a tar file
+"""
+
+import os
+import pprint
+import sh
+import re
+import json
+import base64
+import textwrap
+import subprocess
+import hashlib
+
+from pathlib import Path
+import sys
+from io import StringIO
+from contextlib import redirect_stderr, redirect_stdout
+from ploigos_step_runner import StepImplementer, StepResult, WorkflowResult
+from ploigos_step_runner.utils.io import create_sh_redirect_to_multiple_streams_fn_callback
+from ploigos_step_runner.utils.io import TextIOIndenter
+from ploigos_step_runner.utils.dict import deep_merge
+
+
+
+DEFAULT_CONFIG = {
+    'rekor-server': 'http://rekor.apps.tssc.rht-set.com',
+    'gpg-key': '/var/pgp-private-keys/gpg_public_key',
+    'gpg-user': 'tssc-service-account@redhat.com'
+}
+
+REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS = [
+    'rekor-server',
+    'gpg-key',
+    'gpg-user'
+]
+
+
+class Rekor(StepImplementer):  # pylint: disable=too-few-public-methods
+    """`StepImplementer` for the `automated-governance` step using Rekor.
+    """
+
+    @staticmethod
+    def step_implementer_config_defaults():
+        """Getter for the StepImplementer's configuration defaults.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+
+        """
+        return DEFAULT_CONFIG
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    @staticmethod
+    def base64_encode(
+            file_path
+    ):
+        """Given a file_path, read and encode the contents in base64
+        Returns
+        -------
+        Base64Contents
+            base64 encoded string of file contents
+        """
+        encoding = Path(file_path).read_text().encode('utf-8')
+        return base64.b64encode(encoding).decode('utf-8')
+
+    def get_file_hash(self, file_path):
```

Comment:
this should move to the utils/files.py class

Reply:
https://github.com/ploigos/ploigos-step-runner/pull/146/commits/4c2422a2f700845f352d0ed810e54114acfaa1b3